antarest/core/tasks/service.py

```
@@ -377,6 +371,17 @@ def _cancel_task(self, task_id: str, dispatch: bool = False) -> None:
                 )
             )
 
+    def _do_cancel_task(self, task_id: str) -> None:
+        task = self.repo.get_or_raise(task_id)
+        if task_id in self.tasks:
+            cancelled = self.tasks[task_id].cancel()
```

If `self.tasks[task_id].cancel()` returns False, this function just silently fails ?

---

Well it's a no-op indeed.
It may be relevant to raise in that case so that the API does not return a 2**, good point.
Not sure what would be the best status code, maybe "409 Conflict" ?

---

I never used this but yes it seems appropriate.

---

I propose to leave it as is just because it will be "difficult" to implement:
since the actual cancellation will not be executed by the worker which received the HTTP request, the request is actually an asynchronous one today. We don't wait for the result to check if it has indeed been cancelled.

I will just add a log.